cpp/include/cuml/neighbors/knn.hpp

```diff
@@ -44,8 +44,9 @@ struct knnIndex {
   faiss::gpu::GpuIndex *index;
   int device;
   ~knnIndex() {
-    delete gpu_res;
     delete index;
+    gpu_res->noTempMemory();
```

Comment:
Is this noTempMemory necessary? it seems like reordering the index and gpu_res makes a lot of sense, but I'm not sure noTempMemory is needed

Reply:
Yes, I think I misinterpreted the noTempMemory call (saw it in an issue). I'll fix this and test multiple times.